src/realm/object-store/sync/app.hpp

```diff
@@ -396,20 +432,26 @@ class App : public std::enable_shared_from_this<App>,
     // reloads an app (#5411).
     static void close_all_sync_sessions();
 
-private:
-    friend class Internal;
-    friend class OnlyForTesting;
+    // Return the base url path used for HTTP AppServices requests
+    std::string get_hostname();
+
+    // Return the base url path used for Sync Session Websocket requests
+    std::string get_ws_hostname();
 
+private:
     Config m_config;
 
     // mutable to allow locking for reads in const functions
     // this is a shared pointer to support the App move constructor
     mutable std::shared_ptr<std::mutex> m_route_mutex = std::make_shared<std::mutex>();
-    std::string m_base_url;
+    std::string m_base_url; // Original base URL for server, prior to location request
```

Comment:
it would be nice to have an example url and what each of these refers to in a comment somewhere